docs/sql-migration-guide.md

```diff
@@ -24,6 +24,7 @@ license: |
 
 ## Upgrading from Spark SQL 3.3 to 3.4
   
+  - Since Spark 3.4, INSERT INTO commands with explicit column lists comprising fewer columns than the target table will automatically add the corresponding default values for the remaining columns (or NULL for any column lacking an explicitly-assigned default value). In Spark 3.3 or earlier, these commands would have failed returning errors reporting that the number of provided columns does not match the number of columns in the target table.
```

Comment:
Let's mention users can disable `spark.sql.defaultColumn.useNullsForMissingDefaultValues` to restore the previous behavior?

Reply:
Good idea, done!